tests/test_git_colocated.rs

```diff
@@ -235,3 +235,104 @@ fn test_git_colocated_conflicting_git_refs() {
 fn get_log_output(test_env: &TestEnvironment, workspace_root: &Path) -> String {
     test_env.jj_cmd_success(workspace_root, &["log", "-T", "commit_id \" \" branches"])
 }
+
+#[test]
+fn test_git_colocated_unreachable_commits() {
+    let test_env = TestEnvironment::default();
+    let workspace_root = test_env.env_root().join("repo");
+    let git_repo = git2::Repository::init(&workspace_root).unwrap();
+
+    // Create an initial commit in Git
+    std::fs::write(workspace_root.join("file"), "contents").unwrap();
+    git_repo
+        .index()
+        .unwrap()
+        .add_path(Path::new("file"))
+        .unwrap();
+    let tree1_oid = git_repo.index().unwrap().write_tree().unwrap();
```

Comment:
Optional: We don't care about the contents here, so you can simplify this by doing `let empty_tree_oid = git_repo.treebuilder(None).unwrap().write().unwrap();`. Then use that for both commits.

Reply:
True. Done.